src/libAtomVM/context.c

```diff
@@ -202,14 +194,23 @@ size_t context_size(Context *ctx)
         + context_memory_size(ctx) * BYTES_PER_TERM;
 }
 
+static inline size_t context_heap_size(const Context *ctx)
+{
+    size_t heap_size = ctx->heap.heap_ptr - ctx->heap.heap_start;
```

Comment:
In general my feeling is that we should not be referencing internal elements of heap outside of the memory module.  It might be a bit excessive to define a new heap module, but maybe app heap internal data structures into the memory module?

Reply:
Plus this code is wrong.